llama_index/evaluation/retrieval/metrics_base.py

```diff
@@ -49,3 +49,33 @@ def compute(
             **kwargs: Additional keyword arguments
 
         """
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class BaseIndexlessRetrievalMetric(BaseModel, ABC):
```

Comment:
wasn't obvious to me the difference from `BaseRetrievalMetric`

Reply:
would prefer to not introduce another concept if possible.

Reply:
That's fair. The difference is that `BaseRetrievalMetric` requires `expected_id` and `retrieved_id` whereas there are some metrics, like relevancy that work with texts, rather than ids. I thought it might've been good to represent these separates cases using different abstractions/concepts.

But, I hear you that it might not really be necessary for this case, and so preference would be to not introduce new concepts when we don't absolutely need to.

I'll try re-working this now.

Reply:
Alright, I made a change to `BaseRetrievalMetric` so that it takes in the required inputs for computing cohere-rerank. I also added `expected_texts` in case there is a metric we want to build that uses this i.e., our context similarity metric that we use for benchmarking `LabelledRagDatasets`